File: Website_Phim/WebPhim-Backend/MovieWebsite/MovieWebsite/src/main/java/com/project/MovieWebsite/services/MovieTypeService.java

package com.project.MovieWebsite.services;

import com.project.MovieWebsite.dtos.MovieTypeDTO;
import com.project.MovieWebsite.dtos.OrderDTO;
import com.project.MovieWebsite.models.MovieType;
import com.project.MovieWebsite.models.Order;

import java.util.List;

public interface MovieTypeService {

    MovieType createMovieType(MovieTypeDTO movieTypeDTO);

    MovieType getMovieTypeById(int id);

    List<MovieType> getAllMovieType();

    MovieType updateMovieType(int id, MovieTypeDTO movieTypeDTO);

    void deleteMovieType(int id);
}

File: MovieWebsite-fontend/src/app/services/movie_type.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { MovieType } from '../models/movie_type';
import { environment } from '../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class MovieTypeService {
  private apiGetMovieTypes = `${environment.apiBaseUrl}/movie_types`;

  constructor(private http: HttpClient) { }

  getMovieTypes(): Observable<MovieType[]> {
    return this.http.get<MovieType[]>(this.apiGetMovieTypes);
  }

  // Phương thức để lấy thông tin về một thể loại phim cụ thể dựa trên ID
  getMovieTypeById(movieTypeId: number): Observable<MovieType> {
    const url = `${this.apiGetMovieTypes}/${movieTypeId}`;
    return this.http.get<MovieType>(url);
  }

  // Phương thức để tạo một thể loại phim mới
  createMovieType(movieType: MovieType): Observable<MovieType> {
    return this.http.post<MovieType>(this.apiGetMovieTypes, movieType);
  }

  // Phương thức để cập nhật thông tin của một thể loại phim đã tồn tại
  updateMovieType(movieType: MovieType): Observable<MovieType> {
    const url = `${this.apiGetMovieTypes}/${movieType.id}`;
    return this.http.put<MovieType>(url, movieType);
  }

  // Phương thức để xóa một thể loại phim dựa trên ID
  deleteMovieType(movieTypeId: number): Observable<any> {
    const url = `${this.apiGetMovieTypes}/${movieTypeId}`;
    return this.http.delete<any>(url);
  }
}
